analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -292,9 +316,9 @@ private boolean isOnNewDay(ActivityPayload payload, ActivityDto lastRegisteredAc
 				.isAfter(lastRegisteredActivity.getStartTime());
 	}
 
-	private boolean precedesLastRegisteredActivity(ActivityPayload payload, ActivityDto lastRegisteredActivity)
+	private boolean precedesLastRegisteredActivity(ActivityPayload payload, Optional<ActivityDto> lastRegisteredActivity)
```

Comment:
Maybe rename to `completelyPrecedesLastRegisteredActivity` for clarity now.

Reply:
You can't partially precede something. Then you would only start earlier, but not precede (precede = voorafgaan).

Reply:
OK, I think you're right